cookies/views.py

```diff
@@ -1180,40 +1180,49 @@ def resource_content(request, resource_id):
         return HttpResponseRedirect(target)
     return HttpResponse('Nope')
 
-def export_coauthor_data(request):
+def export_coauthor_data(request, parent_id):
+    """
+    Exporting coauthor data from a collection detail view
+    """
+
     context = RequestContext(request, {})
 
-    graph = nx.Graph()
+    if parent_id:
+        # Prefetching resources and relations from collection db
+        collection = Collection.objects.prefetch_related('native_resources__relations_from').get(id=parent_id)
+        for resource in collection.native_resources.all():
+            relations = resource.relations_from.prefetch_related('predicate','target')
 
-    parent_id = request.GET.get('parent_collection', None)
+        # check if user has permission to view the collection
+        if not authorization.check_authorization('view_collection', request.user, collection):
+            return HttpResponse('You do not have permission to view this collection', status=401)
 
+        # Obtaining id and name of all authors in that collection
+        list_authors = []
+        for relation in relations:
+            if relation.predicate.name == 'Authors':
+                list_authors.append([relation.target.id, relation.target.name])
 
-    if parent_id:
-        parent_collection = _get_collection_by_id(request, int(parent_id))
-        check_auth = authorization.check_authorization('view_collection', request.user, parent_collection)
-        if not check_auth:
-            return HttpResponse('You do not have permission to view this collection', status=401)
+        # creating a complete graph as all authors in a collection are related
```

Comment:
No, that's not correct. From the story: 

> What we want is a graph with concepts (of people) as nodes, and edges indicating that two people appear as authors on the same paper. 

Not all authors in a collection are related -- only those associated with the same paper(s).